.github/workflows/cache.yml

```diff
@@ -0,0 +1,58 @@
+# This workflow caches crate dependencies and build artifacts for tools (except 'test-tools' since we don't use them in build workflows).
+# The cache is only usable by workflows started from pull requests against the develop branch.
+name: CacheDepsAndTools
```

Comment:
I'm not following why we need an entirely separate workflow for creating caches? Isn't this redundant since the build workflow (which runs on PRs against the develop branch and covers the same paths as this workflow) has it's own steps for creating / accessing caches?

Reply:
PR build workflows can't share caches. Merged PR's cache can't be re-used by other PRs even if it goes to develop. However, you can use caches from the default branch (develop). This is a major point of confusion with github action caches. https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#restrictions-for-accessing-a-cache

Reply:
As for why we generate a cache for develop, we want the inital build workflow by a fresh PR to be able to use cached tools so it doesn't have to suffer the 8 - 10 min penalty in building it and creating it's own cache.

Reply:
Also see https://github.com/actions/cache/issues/843#issuecomment-1179771256